Update project to Qt5 and CMake 3.5; README update

Mappaschreck opened the pull request:
- Upgraded project from Qt4 to Qt5.
- Set minimum CMake version to 3.5.
- Updated README to recommend 'libusb-dev' instead of 'libusb'.

Mappaschreck committed:
Update project to Qt5 and CMake 3.5; README update

- Upgraded project from Qt4 to Qt5.
- Set minimum CMake version to 3.5.
- Updated README to recommend 'libusb-dev' instead of 'libusb'.

Mappaschreck committed:
Support DS1490F device, migrate ds9490.cpp to Qt5, increase logging level

- Updated ds9490.cpp to work with Qt5 for broader compatibility
- Enhanced logging level for improved debugging and error tracking

kkoop commented on ds9490.cpp:
```diff
@@ -76,8 +77,8 @@ bool DS9490::AquireUsb(struct usb_device* dev)
-   if (usb_set_configuration(m_usbDevHandle, 1)!=0)
-   {
+   if (usb_set_configuration(m_usbDevHandle, 1) != 0) {
+      std::cerr << "usb_set_configuration error: " << usb_strerror() << std::endl;
```
I would prefer not directly outputting to stderr from this class, instead putting the error message into m_lastError which can then be either printed to stderr or into a QMessageBox by the caller

Mappaschreck replied:
Thanks for the feedback! I have adjusted the error handling and added additional information about usb_strerror().

kkoop commented on gui/mainwindow.cpp:
```diff
@@ -138,8 +138,8 @@ void MainWindow::onReadData()
       QDateTime addedTime=timeStamp.addSecs(i*sampleRate);
-      QString dateLocaleTime = addedTime.date().toString(Qt::SystemLocaleShortDate)
-                              +" "+addedTime.time().toString(Qt::TextDate);      
+ QString dateLocaleTime = QLocale().toString(addedTime.date(), QLocale::ShortFormat)
```
indentation looks off here

Mappaschreck replied:
Thank you for pointing this out; I have corrected the passage.

Mappaschreck committed:
Improvements to mainwindow.cpp and ds9490.cpp:
- Corrects the display of date and time formats in the table (mainwindow.cpp)
- Added error handling in ds9490.cpp to log USB configuration errors more clearly

Mappaschreck committed:
Update README.md

- QT5 instead of QT4
- GnuPlot as a requirement
